Mathlib/Topology/Algebra/InfiniteSum/Constructions.lean

```diff
@@ -73,6 +73,33 @@ section ContinuousMul
 
 variable [CommMonoid α] [TopologicalSpace α] [ContinuousMul α]
 
+section Sum
+
+@[to_additive]
+lemma HasProd.sum {α β M : Type*} [CommMonoid M] [TopologicalSpace M] [ContinuousMul M]
+    {f : α ⊕ β → M} {a b : M}
+    (h₁ : HasProd (f ∘ Sum.inl) a) (h₂ : HasProd (f ∘ Sum.inr) b) : HasProd f (a * b) := by
+  have : Tendsto ((∏ b ∈ ·, f b) ∘ sumEquiv.symm) (atTop.map sumEquiv) (nhds (a * b)) := by
+    rw [Finset.sumEquiv.map_atTop, ← prod_atTop_atTop_eq]
+    convert (tendsto_mul.comp (nhds_prod_eq (x := a) (y := b) ▸ Tendsto.prod_map h₁ h₂))
+    ext s
+    simp
+  simpa [Tendsto, ← Filter.map_map] using this
+
+@[to_additive]
+lemma Sum.tprod {α β M : Type*} [CommMonoid M] [TopologicalSpace M] [ContinuousMul M] [T2Space M]
```

Comment:
it looks to me like tprod_sum might be a closer name to the others in this file, is that available?

Reply:
Yes but the additive version `tsum_sum` is not.

Reply:
Hmm, that one's part of a pair tprod_of_prod / tsum_sum, which doesn't quite make sense. I'd say both of those should mention finset somehow, since they're about Finset.prod and Finset.sum, whereas these ones don't.